lib/go_fish/player.ex

```diff
@@ -124,9 +90,32 @@ defmodule GoFish.Player do
     List.foldl(cards, state, & add_card(&2, &1))
   end
 
+  def go_fish(state) do
+    IO.puts("They didn't have the requested value so I go fishing")
+    case GoFish.Ocean.take_card() do
+        {:card, card} ->
+          if state.hand == [] do
+          end
+          IO.puts("I drew the card #{inspect(card)} from the ocean")
+          {:reply, :went_fishing, %{add_card(state,card) | :is_my_turn => false}}
+        :no_cards_left ->
+          IO.puts("There are no cards left in the ocean")
+          {:reply, :no_cards_left, state}
+    end
+  end
```

Comment:
In this case we had a good reason for it. But just as a note, we should be careful about moving functions around, since it makes it much harder to review, since git can't compare the differences, when it has been moved.